### PR TITLE
[stable/kube-slack] Make kube-slack compatible with EKS 1.16

### LINKED ISSUE #22582: [stable/kube-slack] Make EKS 1.16 compatible
**Describe the bug**
The deployment version is hardcoded. It should either be updated or adopt to the EKS Version.

**Version of Helm and Kubernetes**:
1.16+ 

**Which chart**:
kube-slack

### PR DESCRIPTION
**Pull request by Alien2150**
- fixes https://github.com/helm/charts/issues/22582

### CODE REVIEW

File: stable/kube-slack/templates/deployment.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: apps/v1beta1
+apiVersion: {{ .Values.deployment.apiVersion }}
```
Comment (cpanato) [inline stable/kube-slack/templates/deployment.yaml]: why not just set `apps/v1`?

Comment (Alien2150) [reply stable/kube-slack/templates/deployment.yaml]: Wanted to keep it flexible. In case someone is running EKS 1.13,1.14 or 1.15

Comment (Alien2150) [reply stable/kube-slack/templates/deployment.yaml]: For EKS 1.16 this Value needs to be overwritten. But if you like i can specificy it automatically based on EKS Server version @cpanato

Comment (Alien2150) [reply stable/kube-slack/templates/deployment.yaml]: Determined automatically @cpanato

File: stable/kube-slack/templates/_helpers.tpl [hunk 2]
```diff
@@ -52,3 +52,14 @@ Return the appropriate apiVersion for RBAC APIs.
...
+{{/*
+Return the appropriate apiVersion for deployment.
+*/}}
+{{- define "kube-slack.deployment.apiVersion" -}}
+{{- if semverCompare "<1.9-0" .Capabilities.KubeVersion.GitVersion -}}
+{{- print "extensions/v1beta1" -}}
+{{- else if semverCompare "^1.9-0" .Capabilities.KubeVersion.GitVersion -}}
+{{- print "apps/v1" -}}
+{{- end -}}
+{{- end -}}
```
Comment (cpanato) [inline stable/kube-slack/templates/_helpers.tpl]: nit: new line

Comment (Alien2150) [reply stable/kube-slack/templates/_helpers.tpl]: adding

**Comment by cpanato** [comment]
last request @Alien2150 can you squash the commits?

**Comment by Alien2150** [comment]
@cpanato thx for the feedback. Should be 1 commit now :)

**Comment by Alien2150** [comment]
And news? @cpanato it's already 1 week ago :)

**Comment by stravze** [comment]
Any news on this please ?

**Review by cpanato — APPROVED** [review]
/approve
/lgtm

**Pull request merged**